File: portal/portal/App_Code/Menu.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace portal.App_Code
{
    public class Menu
    {
        private double _codigo;
        private String _nome;
        private Tipos _tipo;
        private TipoStatus _status;
        private List<MenuItem> _items = new List<MenuItem>();

        public double Codigo
        {
            get
            {
                return _codigo;
            }

            set
            {
                _codigo = value;
            }
        }

        public string Nome
        {
            get
            {
                return _nome;
            }

            set
            {
                _nome = value;
            }
        }

        public Tipos Tipo
        {
            get
            {
                return _tipo;
            }

            set
            {
                _tipo = value;
            }
        }

        public TipoStatus Status
        {
            get
            {
                return _status;
            }

            set
            {
                _status = value;
            }
        }

        public List<MenuItem> Items
        {
            get
            {
                return _items;
            }

            set
            {
                _items = value;
            }
        }

        public enum TipoStatus { Inativo, Ativo };
        public enum Tipos { Principal, MenuLateral, HotSite };
    }
}

File: portal/portal/App_Code/Menu.php
<?php
###################################################################################################################
//CADASTROS DE MENUS (menu arvore).
####################################################################################################################

namespace portal\App_Code {
require_once("MenuItem.php");
    
    class Menu {
//ATRIBUTOS----------------------------------------------------------------------------------------------------------
        private $codigo;
        private $nome;
        private $Tipos;// 1-  Principal, 2- MenuLateral, 3- HotSite
        private $TipoStatus;// 1 - Ativo, 0 - Inativo.
		private $items = [];//Lista de Objeto MenuItem
//CONSTRUTOR----------------------------------------------------------------------------------------------------------			
		public function __construct(){}

        public function setMenu($codigo,$nome,$TipoStatus,$tipos){
			$this -> codigo = $codigo;
			$this -> nome = $nome;
			$this -> TipoStatus = $TipoStatus;
			$this -> Tipos = $tipos;
		}        

        public function setMenuComLista($codigo,$nome,$TipoStatus,$tipos,$menuItem){
			$this -> codigo = $codigo;
			$this -> nome = $nome;
			$this -> TipoStatus = $TipoStatus;
			$this -> Tipos = $tipos;
			$this -> items= (array)$menuItem;
		}
//SETS-----------------------------------------------------------------------------------------------------------------        
		public function setCodigo($codigo){
			$this -> codigo = $codigo;
        }
        public function setNome($nome) {
			$this -> nome = $nome;
        }
        public function setTipoStatus($TipoStatus) {
			$this -> TipoStatus = $TipoStatus;
        }
        public function setTipos($tipos) {
			$this -> Tipos = $tipos;
        }
        public function setItems($MenuItem) {
			$this -> items[] = (array)$MenuItem;
        }
//GETS-----------------------------------------------------------------------------------------------------------------			
		public function getCodigo(){
			return $this -> codigo;
        }
        public function getNome(){
			return $this -> nome;
        }
        public function getTipoStatus() {
			return $this -> TipoStatus;
        }
        public function getTipos() {
			return $this -> Tipos;
        }		
        public function getItems() {
			return $this -> items;
        }		
    }
}
?>